### ISSUE TITLE
[a11y Support] MHV Appointment - Appointment details link's aria-label may pose challenges for voice control users

### ISSUE THREAD

**Issue by rileyorr** [department-of-veterans-affairs/va.gov-team#94246]
> [!NOTE]
> **QUICK SUMMARY:** The MHV Appointment details link may pose challenges for folks using voice control due to it using the aria-label tag, which historically has resulted in difficulties when navigating with voice control.

## Issue Description
During a discussion with Bobby, he mentioned that he had noticed that the 'details' links associated with MHV appointments are using the aria-label tag to describe the appointment since there can be multiple appointments with the same 'details' link content. However, aria-label tags sometimes present challenges when navigating using voice control because they require the user to use specific verbiage that isn't visible on the page. 

This ticket's purpose is to test the experience to see if it's navigable with voice control and reach out to the Appointments team to learn if they've conducted research on the design with voice control users. We may want to make a recommendation based on our findings. 

## About the Team

- **Team name**: MHV Appointments
- **Product name:** Appointments

### Supporting Artifacts 
- ![A screenshot of a list of past appointments. Appointments are displayed in a table format and include information about the location, date, time, and appointment type. Each appointment also has a details link that takes the user to another page with more details about the appointment.](https://github.com/user-attachments/assets/3edd2206-c40a-4222-9e47-ac6fbf23288f)

```[tasklist]
### a11y Tasks
- [x] Test the experience using voice control
- [x] Document findings in this ticket
- [x] Reach out to the Appointments team in Slack to learn more about any AT research conducted on this page
- [x] Document any recommendations and next steps discussed with the team in this ticket
```
```[tasklist]
### Acceptance Criteria
- [x] Findings and recommendations are documented in this ticket and communicated with the Appointments team
```
```[tasklist]
## Ticket Updates (CAIA Internal)
- [ ] _Connect to an `Epic` or `Intake` (what body of work is this a part of?)_
- [ ] _Label with `Originator/Team` (product team or stakeholder requesting support)_
- [ ] _Label date in the `Open Date` field_
- [ ] _Label with `Estimate` (level of effort expected for this ticket)_
- [ ] _Add `Assignee(s)` name(s) to ticket_ (include Naomi)
- [ ] _Add `Assignee(s) name(s) to each task` they will complete via handle tag (if known)_
- [ ] _Select a `Priority Level`_
- [ ] _Update date in `Last Checked` field_
- [ ] _Label with `Actual` (level of effort it took to complete this ticket)_
- [ ] _Update date in_ `Closed Date`
```

```[tasklist]
### Related Tickets
- [ ] TBD
```

**Comment by rileyorr** [comment]
FYI I started this work yesterday so I can move it into 'In progress' once the intake is reviewed. I'll write my initial findings in a subsequent comment

**Comment by rileyorr** [comment]
I tested the experience out in Safari using Voice Control and noticed a few things: 
- Voice control does not play well with aria-label 
- I can select elements by their number (i.e. 'Show numbers' and selecting a number), but I cannot select elements that are using aria-label (ex. 'Click Print') 
   - So technically everything is selectable using Voice Control, but we can't assume we know everyone's setup or that everyone knows they can select elements using numbers
- I can select the print button when using the aria-label 'Print list', but that may not be inherently understood by Voice Control users that they need to search for the aria-label and use that to select the element 
- The aria-labels for the appointment details links are very long, and when I attempted to select a link similarly to how I selected the print button I was unable to 

I'm going to start a thread with the Appointments team and include a screen recording of what I experienced for reference. They may have done research on this page with Voice Control users. Bobby also mentioned another team ran into this issue in the past so I am going to tag him to learn more about what they did to resolve this issue. We may want to elevate this to the governance/platform team for Brian's feedback.

**Comment by rileyorr** [comment]
Started [a thread in the Appointment's UX channel](https://dsva.slack.com/archives/C03CGTDLTFF/p1727984103633899) with my thoughts and tagged Bobby as well for his take. We're waiting for a response from the team, but our recommendation would be to switch the aria-label tags to aria-describedby.

**Comment by rileyorr** [comment]
I met with the team yesterday afternoon to chat about the recommendations and answer questions. They're going to chat with product to prioritize the fix and reach out to me for further help once the design team connects with dev.

**Comment by sara-amanda** [comment]
Closing with comment to match GHP status. Closed in OCTO/a11y call on 10/8/2024.
